cmd/baseserver/main.go

```diff
@@ -0,0 +1,67 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+
+	"github.com/gorilla/mux"
+	"github.com/spf13/cobra"
+
+	"github.com/tendermint/basecoin/client/commands"
+	rest "github.com/tendermint/basecoin/client/rest"
+	keys "github.com/tendermint/go-crypto/keys"
+	"github.com/tendermint/tmlibs/cli"
+)
+
+var (
+	keysManager keys.Manager
+)
+
+func init() {
+	keysManager = rest.DefaultKeysManager()
+}
+
+var srvCli = &cobra.Command{
+	Use:   "baseserver",
+	Short: "Light REST client for tendermint",
+	Long:  `Baseserver presents  a nice (not raw hex) interface to the basecoin blockchain structure.`,
+}
+
+var serveCmd = &cobra.Command{
+	Use:   "serve",
+	Short: "Serve the light REST client for tendermint",
+	Long:  "Access basecoin via REST",
+	RunE:  serve,
+}
+
+func main() {
+	commands.AddBasicFlags(srvCli)
+	srvCli.AddCommand(
+		commands.InitCmd,
+		serveCmd,
+	)
+
+	// TODO: Decide whether to use $HOME/.basecli for compatibility
+	// or just use $HOME/.baseserver?
+	cmd := cli.PrepareMainCmd(srvCli, "BC", os.ExpandEnv("$HOME/.basecli"))
+	if err := cmd.Execute(); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func serve(_ *cobra.Command, _ []string) error {
+	port := 8998
+	router := mux.NewRouter()
+	ctx := rest.Context{
+		Keys: rest.New(keysManager, ""),
```

Comment:
You can safely set the default here as "ed25519".

We use it like 97% of the time...  secp256k1 is mainly used to make sure other algorithms work